lisa/_assets/kmodules/sched_tp/tp.c

```diff
@@ -95,6 +95,18 @@ static void sched_pelt_irq_probe(struct feature *feature, struct rq *rq)
 }
 DEFINE_TP_EVENT_FEATURE(sched_pelt_irq, pelt_irq_tp, sched_pelt_irq_probe);
 
+static void sched_pelt_thermal_probe(struct feature *feature, struct rq *rq)
+{
+	const struct sched_avg *avg = sched_tp_rq_avg_thermal(rq);
+	int cpu = sched_tp_rq_cpu(rq);
+
+	if (!avg)
+		return;
+
+	trace_sched_pelt_thermal(cpu, avg);
+}
+DEFINE_TP_EVENT_FEATURE(sched_pelt_thermal, pelt_irq_tp, sched_pelt_thermal_probe);
```

Comment:
Could use a comment as to why the probe is attached to that irq related tp. Is it fundamentally looking at an irq-only signal or is it just a convenient tracepoint triggered "often enough" or something like that ?

Reply:
This is a copy&paste error. It didn't show up as an issue while testing it since the `tp_name` parameter of DEFINE_TP_ENABLE_DISABLE() is only there for testing whether a trace_point with this name exists. And `pelt_irq_tp` existed so everything went fine enabling the pelt_thermal_tp related trace_event. Has to be corrected to use `pelt_thermal_tp` instead of `pelt_irq_tp`. thermal and irq are independent EPLT trace signals, like dl and rt as well.

Reply:
The tracepoint is not just checked it it exists, it's also the one that the probe is attached as well but if they have the same signature that code must have "worked" in your test